addons/sale/models/sale_order.py

```diff
@@ -405,6 +405,8 @@ def _compute_user_id(self):
         for order in self:
             if not order.user_id:
                 order.user_id = order.partner_id.user_id or order.partner_id.commercial_partner_id.user_id or self.env.user
+            elif order.user_id != order.partner_id.user_id:
+                order.user_id = order.partner_id.user_id or order.user_id
```

Comment:
```suggestion
            user = order.partner_id.user_id or order.partner_id.commercial_partner_id.user_id
            if not order.user_id or order.user_id != user:
                 order.user_id = user or self.env.user
```
this seems simpler, and take into account `order.partner_id.commercial_partner_id.user_id` might be the one to set the salesperson

Reply:
Thanks for the review 😄 You're right, much simpler to avoid unnecessary if statements.